gossip/hub/configpb/config.proto

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package configpb;
+
+import "github.com/google/trillian/crypto/keyspb/keyspb.proto";
+import "github.com/google/trillian/crypto/sigpb/sigpb.proto";
+import "google/protobuf/any.proto";
+
+// HubMultiConfig wraps up a HubBackendSet and corresponding HubConfigSet so
+// that they can easily be parsed as a single proto.
+message HubMultiConfig {
+  // The set of backends that this configuration will use to send requests to.
+  // The names of the backends in the HubBackendSet must all be distinct.
+  HubBackendSet backends = 1;
+  // The set oubs that will use the above backends. All the protos in this
+  // HubConfigSet must set a valid hub_backend_name for the config to be usable.
+  HubConfigSet hub_configs = 2;
+}
+
+// HubBackendSet describes the set of Trillian backends that support a gossip hub.
+message HubBackendSet {
+  repeated HubBackend backend = 1;
+}
+
+// HubBackend describes a single Trillian backend.
+message HubBackend {
+  // name defines the name of the hub backend for use in HubConfig messages and must be unique.
```

Comment:
If it must be unique, should this be a map?

Reply:
Yeah, that makes life significantly simpler - done.